core/src/main/java/com/linecorp/armeria/common/multipart/DefaultMultipart.java

```diff
@@ -197,6 +206,24 @@ public String toString() {
                           .toString();
     }
 
+    @SuppressWarnings("unchecked")
+    private static <T extends HttpHeaders> T injectBoundary(String boundary, T headers) {
+        requireNonNull(headers, "headers");
+        @Nullable
+        MediaType contentType = headers.contentType();
+        if (contentType != null) {
+            checkArgument(contentType.isMultipart(),
+                          "Content-Type: %s (expected: multipart content type)", contentType);
+            contentType = contentType.withParameter(BOUNDARY_PARAMETER, boundary);
+        } else {
+            contentType = DEFAULT_MULTIPART_TYPE.withParameter(BOUNDARY_PARAMETER, boundary);
+        }
+        final MediaType finalMediaType = contentType;
+        return (T) headers.withMutations(builder -> {
+            builder.setObject(HttpHeaderNames.CONTENT_TYPE, finalMediaType);
+        });
```

Comment:
nit:
```suggestion
        return (T) headers.withMutations(builder -> builder.contentType(finalMediaType));
```

Reply:
Thought about this too. But ended up simply replacing `addObject` to `setObject` in the existing code fragment. Let me check it quickly